File: Dataset/Leetcode/train/101/172.java
 class Solution {
    public boolean XXX(TreeNode root) {
        if(root == null){
            return true;
        }
        return this.compare(root.left,root.right);
    }
    public boolean compare(TreeNode left,TreeNode right){
        if(left==null && right!=null){
            return false;
        }else if(left !=null && right == null){
            return false;
        }else if(left == null && right == null){
            return true;
        }else if(left.val != right.val){
            return false;
        }
        boolean isSide = compare(left.left,right.right);
        boolean isOuter = compare(left.right,right.left);
        return isOuter && isSide;
    }
}

File: Dataset/Leetcode/test/104/172.cpp
class Solution {
public:
    int count = 0 ;
    int XXX(TreeNode* root) {
        if(root == nullptr) return count ;
        queue<TreeNode*> q ;
        q.push(root) ;
        while(!q.empty()){
            count ++ ;
            for(int i = q.size() ; i ;i --){
                auto node = q.front() ;
                q.pop() ;
                if(node -> left != nullptr) q.push(node -> left) ;
                if(node -> right != nullptr) q.push(node -> right) ;
            }
        }
        return count ;
    }
};


